### PR TITLE
SONAR-7316 use String.intern in ChangeSet + fix DBScmInfo

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/scm/Changeset.java [hunk 1]
```diff
@@ -109,26 +109,20 @@ public boolean equals(Object o) {
...
 
   @Override
   public String toString() {
     return "Changeset{" +
-        "revision='" + revision + '\'' +
-        ", author='" + author + '\'' +
-        ", date=" + date +
-        '}';
+      "revision='" + revision + '\'' +
+      ", author='" + author + '\'' +
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/scm/Changeset.java]: `author` can be null so it can display `author='null'`

**Comment by simonbrandhof** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/computation/scm/Changeset.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/computation/scm/Changeset.java]: we rarely bother handling this case in `toString()`

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**